core-domain/src/test/java/org/uniprot/core/uniparc/impl/UniParcCrossReferenceBuilderTest.java

```diff
@@ -43,77 +45,87 @@ void testActive() {
-                        .id("P12345")
-                        .version(7)
-                        .active(true)
-                        .created(created)
-                        .lastUpdated(lastUpdated)
-                        .build();
-        assertEquals(3, xref.getVersionI());
-        assertEquals(7, xref.getVersion().intValue());
-        assertTrue(xref.isActive());
-        assertEquals(created, xref.getCreated());
+                new UniParcCrossReferenceBuilder().lastUpdated(lastUpdated).build();
         assertEquals(lastUpdated, xref.getLastUpdated());
-        assertEquals(UniParcDatabase.SWISSPROT, xref.getDatabase());
-        assertEquals("P12345", xref.getId());
     }
 
     @Test
     void testProperties() {
-        LocalDate created = LocalDate.of(2017, 5, 17);
-        LocalDate lastUpdated = LocalDate.of(2017, 2, 27);
         List<Property> properties = new ArrayList<>();
-        properties.add(new Property(UniParcCrossReference.PROPERTY_PROTEIN_NAME, "some pname"));
-        properties.add(new Property(UniParcCrossReference.PROPERTY_GENE_NAME, "some gname"));
+        properties.add(new Property("prop1", "some pname"));
+        properties.add(new Property("prop2", "some gname"));
         UniParcCrossReference xref =
-                new UniParcCrossReferenceBuilder()
-                        .versionI(3)
-                        .database(UniParcDatabase.SWISSPROT)
-                        .id("P12345")
-                        .version(7)
-                        .active(true)
-                        .created(created)
-                        .lastUpdated(lastUpdated)
-                        .propertiesSet(properties)
-                        .build();
-        assertEquals(3, xref.getVersionI());
-        assertEquals(7, xref.getVersion().intValue());
-        assertTrue(xref.isActive());
-        assertEquals(created, xref.getCreated());
-        assertEquals(lastUpdated, xref.getLastUpdated());
-        assertEquals(UniParcDatabase.SWISSPROT, xref.getDatabase());
-        assertEquals("P12345", xref.getId());
+                new UniParcCrossReferenceBuilder().propertiesSet(properties).build();
         assertEquals(properties, xref.getProperties());
     }
 
+    @Test
+    void testGeneName() {
+        String geneName = "gene Name value";
+        UniParcCrossReference xref = new UniParcCrossReferenceBuilder().geneName(geneName).build();
+        assertEquals(geneName, xref.getGeneName());
+    }
+
+    @Test
+    void testProteinName() {
+        String proteinName = "protein Name value";
+        UniParcCrossReference xref =
+                new UniParcCrossReferenceBuilder().proteinName(proteinName).build();
+        assertEquals(proteinName, xref.getProteinName());
+    }
+
+    @Test
+    void testTaxonomy() {
+        Organism organism = new OrganismBuilder().taxonId(123L).build();
+        UniParcCrossReference xref = new UniParcCrossReferenceBuilder().taxonomy(organism).build();
```

Comment:
I think, it is good idea to set all the fields of Organism which are being returned in response like scientificName and taxonId

Reply:
Okay, I can do it.. Can you please explain me what difference it will make to test we are doing? sorry, it is not clear for me.